### PR TITLE
[Spark-20775][SQL] Added scala support from_json

### PR DESCRIPTION
**Pull request by rubenljanssen**
## What changes were proposed in this pull request?

from_json function required to take in a java.util.Hashmap. For other functions, a java wrapper is provided which casts a java hashmap to a scala map. Only a java function is provided in this case, forcing scala users to pass in a java.util.Hashmap.

Added the missing wrapper.

## How was this patch tested?
Added a unit test for passing in a scala map

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -3060,7 +3060,7 @@ object functions {
     from_json(e, schema, Map.empty[String, String])
 
   /**
-   * Parses a column containing a JSON string into a `StructType` or `ArrayType` of `StructType`s
+   * (Java-specific) Parses a column containing a JSON string into a `StructType` or `ArrayType` of `StructType`s
```
Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: nit: ScalaStyle check will fail saying `File line length exceeds 100 characters`.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -3072,6 +3072,22 @@ object functions {
    * @since 2.1.0
    */
   def from_json(e: Column, schema: String, options: java.util.Map[String, String]): Column = {
+    from_json(e, schema, options.asScala.toMap)
+  }
+
+  /**
+    * (Scala-specific) Parses a column containing a JSON string into a `StructType` or `ArrayType` of `StructType`s
```
Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: ditto.

**Review by ueshin — COMMENTED** [review]
LGTM except for minor comments.
cc @cloud-fan @gatorsmile

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 3]
```diff
@@ -3072,6 +3072,22 @@ object functions {
...
+    * with the specified schema. Returns `null`, in the case of an unparseable string.
+    *
+    * @param e a string column containing JSON data.
+    * @param schema the schema to use when parsing the json string as a json string. In Spark 2.1,
+    *               the user-provided schema has to be in JSON format. Since Spark 2.2, the DDL
+    *               format is also supported for the schema.
+    *
+    * @group collection_funcs
+    * @since 2.1.0
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: `@since 2.3.0`?

**Comment by cloud-fan** [comment]
ok to test

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**